src/learner.cc

```diff
@@ -779,9 +810,17 @@ class LearnerConfiguration : public Learner {
     if (obj_ == nullptr || tparam_.objective != old.objective) {
       obj_.reset(ObjFunction::Create(tparam_.objective, &ctx_));
     }
+
+    bool has_nc {cfg_.find("num_class") != cfg_.cend()};
```

Comment:
This change is made to fix the removed lines above. The change limits the inserted parameter into this function instead of propagating it through the entire library.